src/AI/JakeMan.java

```diff
@@ -0,0 +1,770 @@
+          Map<XYCoord, Double> threatArea = mapToFill.get(um);
+          double newValue = threat.getHPFactor();
+          // Square unit fraction so low-HP units aren't valued so much
+          newValue *= newValue;
+          for( XYCoord coord : AICombatUtils.findThreatPower(gameMap, threat, null).keySet() )
+          {
+            if( !threatArea.containsKey(coord) )
+              threatArea.put(coord, newValue);
+            else
+              threatArea.put(coord, newValue + threatArea.get(coord));
+          }
+        }
+      }
+
+      return null;
+    }
+  }
+  
+  // Try to get unit value by capture or attack
+  public static class GetFreeDudes extends UnitActionFinder
+  {
+    private static final long serialVersionUID = 1L;
+    private final JakeMan ai;
+    private final boolean canEvict, canStepOnProduction;
+    public GetFreeDudes(Army co, JakeMan ai, boolean canEvict, boolean canStepOnProduction)
+    {
+      super(co, ai);
+      this.ai = ai;
+      this.canEvict = canEvict;
+      this.canStepOnProduction = canStepOnProduction;
+    }
+
+    @Override
+    public GameAction getUnitAction(Unit unit, GameMap gameMap)
+    {
+      boolean mustMove = false;
+      return findFreeDude(unit.CO, ai, unit, gameMap, mustMove, !canStepOnProduction, canEvict);
+    }
+
+    public static GameAction findFreeDude( Commander co, JakeMan ai,
+                                              Unit unit, GameMap gameMap,
+                                              boolean mustMove, boolean avoidProduction,
+                                              boolean canEvict )
+    {
+      XYCoord position = new XYCoord(unit.x, unit.y);
+
+      boolean includeOccupiedSpaces = true; // Since we know how to shift friendly units out of the way
+      ArrayList<XYCoord> destinations = Utils.findPossibleDestinations(unit, gameMap, includeOccupiedSpaces);
+      if( mustMove )
+        destinations.remove(new XYCoord(unit.x, unit.y));
+      destinations.removeAll(AIUtils.findAlliedIndustries(gameMap, co.army, destinations, !avoidProduction));
+      // sort by furthest away, good for capturing
+      Utils.sortLocationsByDistance(position, destinations);
+      Collections.reverse(destinations);
+      ArrayList<GameAction> freeDudeShots = new ArrayList<>();
+
+      for( XYCoord moveCoord : destinations )
+      {
+        // Figure out how to get here.
+        GamePath movePath = Utils.findShortestPath(unit, moveCoord, gameMap);
+
+        // Figure out what I can do here.
+        ArrayList<GameActionSet> actionSets = unit.getPossibleActions(gameMap, movePath, includeOccupiedSpaces);
+        for( GameActionSet actionSet : actionSets )
+        {
+          Unit resident = gameMap.getLocation(moveCoord).getResident();
+          boolean spaceFree = null == resident;
+          if( !spaceFree )//&& ((unit.CO != resident.CO || resident.isTurnOver)) )
+            continue; // Bail if we can't clear the space
```

Comment:
`canEvict` is never used in this function - should we check it here before bailing?

Reply:
It appears I never got around to handling eviction properly here.
I imagine I punted because nothing like that was in the original writeup.

Reply:
Well. I'd advocate to clean this up one way or the other; clean up the unused nonsense, or actually use it. Then again, I don't care enough to make you, so you decide.